app/controllers/posts_controller.rb

```diff
@@ -268,17 +268,11 @@ def search
     end
     @search_results = @search_results.complete if params[:completed].present?
     if params[:author_id].present?
-      post_ids = nil
-      params[:author_id].each do |author_id|
-        author_posts = Post::Author.where(user_id: author_id, joined: true).pluck(:post_id)
-        if post_ids.nil?
-          post_ids = author_posts
-        else
-          post_ids &= author_posts
-        end
-        break if post_ids.empty?
-      end
-      @search_results = @search_results.where(id: post_ids.uniq)
+      # get author matches for posts that have at least one
+      author_posts = Post::Author.where(user_id: params[:author_ids]).group(:post_id)
+      # select posts that have all of them
+      author_posts = author_posts.having('COUNT(post_authors.user_id) = ?', params[:author_ids].length).pluck(:post_id)
```

Comment:
Ooh, this is interesting. If we weren't so reliant on Rails for generating the SQL here, I'd probably do some condition like plucking an array of distinct user IDs and checking it's equal to the ones we want, but I'm having trouble thinking how to translate that into Rails. Since PostAuthor is probably unique enough over (post, user), this should be fine!

Reply:
(I think we can technically `.select(:post_id)` here and use a subquery but I know we're worried about introducing too many of those =P just saying)

Reply:
it should be unique on that scope, yeah, on the rails level if not the db level

Reply:
Oh, we could maybe do `COUNT(DISTINCT post_authors.user_id)` to guarantee this is true even if we don't have a unique constraint over (post, user), but I think we do, so it shouldn't be an issue